pkg/deploy/controller/deployment/controller.go

```diff
@@ -138,18 +138,44 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Nothing to do in this terminal state.
 		glog.V(4).Infof("Ignoring deployment %s (status %s)", deployutil.LabelForDeployment(deployment), currentStatus)
 	case deployapi.DeploymentStatusComplete:
-		// Automatically clean up successful pods.
-		// TODO: Could probably do a lookup here to skip the delete call, but it's
-		// not worth adding yet since (delete retries will only normally occur
-		// during full a re-sync).
+		// this code is maintained for backwards compatibility
+		// we will delete the pod identified by the deployment annotation first
 		podName := deployutil.DeployerPodNameFor(deployment)
-		if err := c.podClient.deletePod(deployment.Namespace, podName); err != nil {
-			if !kerrors.IsNotFound(err) {
-				return fmt.Errorf("couldn't delete completed deployer pod %s/%s for deployment %s: %v", deployment.Namespace, podName, deployutil.LabelForDeployment(deployment), err)
+		// at some point (now or later), we may stop setting this deployer pod name annotation
+		if podName != "" {
```

Comment:
Not sure why we're bothering to keep this logic, since `getDeployerPodsFor` will capture the deployer and hook pods.